src/main/kotlin/org/kamiblue/client/module/modules/combat/CombatSetting.kt

```diff
@@ -131,9 +130,10 @@ internal object CombatSetting : Module(
         }
 
         safeListener<TickEvent.ClientTickEvent>(5000) {
-            for ((function, future) in jobMap) {
-                if (future.isActiveOrFalse) continue // Skip if the previous thread isn't done
-                jobMap[function] = defaultScope.launch { function(this@safeListener) }
+            runBlocking {
+                for (function in taskList) {
+                    defaultScope.launch { function(this@safeListener) }
+                }
             }
```

Comment:
You don't need runBlocking if you are launching it on the defaultScope

Reply:
I explained in the comment in the PR about my reasoning behind changing this from background coroutines to blocking on each update tick.

Alternative also present as a way forward and I can revert this behavior and implement that variation.